ddtrace/tracer/remote_config.go

```diff
@@ -80,11 +114,15 @@ func (t *tags) toMap() *map[string]interface{} {
 
 func (t *tracer) dynamicInstrumentationRCUpdate(u remoteconfig.ProductUpdate) map[string]state.ApplyStatus {
 	applyStatus := map[string]state.ApplyStatus{}
-
 	for k, v := range u {
 		log.Debug("Received dynamic instrumentation RC configuration for %s\n", k)
 		applyStatus[k] = state.ApplyStatus{State: state.ApplyStateUnknown}
-		passFullConfiguration(globalconfig.RuntimeID(), k, string(v))
+
+		diRCState.state[k] = dynamicInstrumentationRCProbeConfig{
+			runtimeID:     globalconfig.RuntimeID(),
+			configPath:    k,
+			configContent: string(v),
+		}
```

Comment:
Shouldn't this assignment or the whole loop be guarded with `diRCState.mu.{,Un}Lock`?

Reply:
Yes, silly mistake after last change.